base/v0_2/translate_test.go

```diff
@@ -1240,9 +1241,15 @@ func TestTranslateTree(t *testing.T) {
 			continue
 		}
 
-		assert.Equal(t, test.outFiles, actual.Storage.Files, "#%d: files mismatch", i)
-		assert.Equal(t, []types.Directory(nil), actual.Storage.Directories, "#%d: directories mismatch", i)
-		assert.Equal(t, test.outLinks, actual.Storage.Links, "#%d: links mismatch", i)
+		if diff := cmp.Diff(test.outFiles, actual.Storage.Files); diff != "" {
```

Comment:
We just switched everything over to testify (#161), which does show diffs and also lets us avoid the explicit comparison.

Reply:
I went ahead and removed the commit that switch to cmp. It's not important to the rest of the change.

For context, early in the implementation there was a bug because [fs.DirEntry.Type()](https://tip.golang.org/pkg/io/fs/#DirEntry) masks out permission bits. I switched to cmp so I could make sense of the test output.


Before (testify):

> ```
> --- FAIL: TestTranslateTree (0.01s)
>     translate_test.go:1243: 
>                 Error Trace:    translate_test.go:1243
>                 Error:          Not equal: 
> ```
> expected: []types.File{types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/overridden", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(0xc0001e78b0)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc0001e78c0), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc000600490)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/overridden-executable", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(0xc0001e78d0)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc0001e78e0), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc000600498)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/executable", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc0001e78f0), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0006004a0)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/file", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc0001e7900), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0006004a8)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/subdir/file", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc0001e7910), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0006004b0)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/subdir/subdir/subdir/subdir/subdir/subdir/subdir/subdir/subdir/file", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(0xc0001e7930), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc0001e7920), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0006004b8)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/etc/file", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc0001e7940), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0006004c0)}}}
> actual  : []types.File{types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/overridden", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(0xc00017ea60)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc00068a1e0), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc000600480)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/overridden-executable", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(0xc00017f050)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc00068a1f0), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc000600488)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/executable", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc00017fd10), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0000a6528)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/file", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc00068a1d0), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0000a6898)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/subdir/file", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc00068a000), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0000a60c8)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/subdir/subdir/subdir/subdir/subdir/subdir/subdir/subdir/subdir/file", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(0xc00068aa00), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc00068a9f0), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0000a66e0)}}, types.File{Node:types.Node{Group:types.NodeGroup{ID:(*int)(nil), Name:(*string)(nil)}, Overwrite:(*bool)(nil), Path:"/etc/file", User:types.NodeUser{ID:(*int)(nil), Name:(*string)(nil)}}, FileEmbedded1:types.FileEmbedded1{Append:[]types.Resource(nil), Contents:types.Resource{Compression:(*string)(nil), HTTPHeaders:types.HTTPHeaders(nil), Source:(*string)(0xc00068ae50), Verification:types.Verification{Hash:(*string)(nil)}}, Mode:(*int)(0xc0000a6a98)}}}
> ```
>                                 Diff:
>                                 --- Expected
>                                 +++ Actual
>                                 @@ -76,3 +76,3 @@
>                                     },
>                                 -   Mode: (*int)(493)
>                                 +   Mode: (*int)(420)
>                                    }
>                 Test:           TestTranslateTree
>                 Messages:       #1: files mismatch
> FAIL
> FAIL    github.com/coreos/fcct/base/v0_2        0.022s
> FAIL
> ```


After (cmp):
```
--- FAIL: TestTranslateTree (0.01s)
    translate_test.go:1245: #1: files mismatch:
          []types.File{
                {Node: {Path: "/overridden", User: {Name: &"bovik"}}, FileEmbedded1: {Contents: {Source: &"data:,tree%2Foverridden"}, Mode: &384}},
                {Node: {Path: "/overridden-executable", User: {Name: &"bovik"}}, FileEmbedded1: {Contents: {Source: &"data:,tree%2Foverridden-executable"}, Mode: &384}},
                {
                        Node: {Path: "/executable"},
                        FileEmbedded1: types.FileEmbedded1{
                                Append:   nil,
                                Contents: {Source: &"data:,tree%2Fexecutable"},
        -                       Mode:     &493,
        +                       Mode:     &420,
                        },
                },
                {Node: {Path: "/file"}, FileEmbedded1: {Contents: {Source: &"data:,tree%2Ffile"}, Mode: &420}},
                {Node: {Path: "/subdir/file"}, FileEmbedded1: {Contents: {Source: &"data:,tree%2Fsubdir%2Ffile"}, Mode: &420}},
                ... // 2 identical elements
          }
FAIL
FAIL    github.com/coreos/fcct/base/v0_2        0.024s
FAIL
```